test/cppsuite/tests/hs_cleanup.cxx

```diff
@@ -59,14 +61,10 @@ class hs_cleanup : public test {
         /* We don't know the keyrange we're operating over here so we can't be much smarter here. */
         while (tc->running()) {
             tc->sleep();
-            ret = cursor->next(cursor.get());
-            if (ret != 0) {
-                if (ret == WT_NOTFOUND) {
-                    testutil_check(cursor->reset(cursor.get()));
-                    continue;
-                } else
-                    testutil_die(ret, "cursor->next() failed unexpectedly.");
-            }
+
+            if (tc->next(cursor) != 0)
```

Comment:
When does this can happen ? Is it possible that we reach a point where it will always return something different than 0 ?

Reply:
Do you mean when can `cursor->next()` return non zero?

Reply:
Yes

Reply:
Reaching the end of the table is the most common case. You get a `WT_NOTFOUND` and then you end up at the beginning of the table again with another `next` call.

Reply:
I think its possible but only if we next over a table without any contents.